Temporal: Move some tests out of round.js under staging into separate files under test/built-ins

catamorphism opened the pull request:
Start breaking up `Temporal/Duration/old/round.js` into separate tests and add those files under `test/built-ins`.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/does-not-balance-up-to-weeks-if-largest-unit-larger-than-weeks.js:
This one looks to be a duplicate of the file above it.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/balances-days-up-to-both-years-and-months.js:
```diff
@@ -0,0 +1,16 @@
+
+TemporalHelpers.assertDuration(twoYears.round({
+    largestUnit: "years",
+    relativeTo: "2017-01-01"
```
This is a common pattern in the staging tests unfortunately, but I'd suggest creating a `relativeTo` object with the PlainDate constructor and passing it in here, to avoid the path through the unrelated string parsing code (here and below)

catamorphism replied:
Fixed

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/rounding-increments.js:
```diff
@@ -0,0 +1,54 @@
+---*/
+
+const d = new Temporal.Duration(5, 5, 5, 5, 5, 5, 5, 5, 5, 5);
+const relativeTo = Temporal.PlainDate.from("2020-01-01");
```
```suggestion
const relativeTo = new Temporal.PlainDate(2020, 1, 1);
```

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/singular-units.js:
```diff
@@ -0,0 +1,156 @@
+---*/
+
+const d = new Temporal.Duration(5, 5, 5, 5, 5, 5, 5, 5, 5, 5);
+const relativeTo = Temporal.PlainDate.from("2020-01-01");
```
Ditto.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/valid-increments.js:
```diff
@@ -0,0 +1,86 @@
+---*/
+
+const d = new Temporal.Duration(5, 5, 5, 5, 5, 5, 5, 5, 5, 5);
+const relativeTo = Temporal.PlainDate.from("2020-01-01");
```
Ditto.

catamorphism replied:
Fixed

ptomato reviewed (changes requested):
Thanks! A few comments mostly about how to avoid testing unrelated paths, but there are a couple about possible improvements.

catamorphism commented on test/built-ins/Temporal/Duration/prototype/round/accepts-datetime-strings-for-relative-to.js:
Yeah, I think I'd rather do it in a followup.

catamorphism commented on test/built-ins/Temporal/Duration/prototype/round/durations-do-not-balance-beyond-largest-unit.js:
```diff
@@ -0,0 +1,14 @@
+
+const fortyDays = new Temporal.Duration(0, 0, 0, 40, 0, 0, 0, 0, 0, 0);
+
+TemporalHelpers.assertDuration(fortyDays.round({ smallestUnit: "seconds" }),
```
Tried doing both in [b4431cc](https://github.com/tc39/test262/pull/4280/commits/b4431cc8764d0bbd4ead538110d44aafc677ca4c) -- let me know if that's what you had in mind.

catamorphism commented:
@ptomato I think I've addressed all your comments (except for creating more tests for `compare`, which I'll try to do in a separate PR) -- let me know if I missed any.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/balances-down-differently-depending-on-relative-to.js:
```diff
@@ -0,0 +1,33 @@
+
+TemporalHelpers.assertDuration(oneYear.round({
+    largestUnit: "days",
+    relativeTo: "2019-01-01"
```
```suggestion
    relativeTo: new Temporal.PlainDate(2019, 1, 1)
```
(and elsewhere in this file)

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/balances-up-differently-depending-on-relative-to.js:
```diff
@@ -0,0 +1,53 @@
+
+TemporalHelpers.assertDuration(fortyDays.round({
+    largestUnit: "years",
+    relativeTo: "2020-01-01"
```
Ditto.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/balances-up-to-next-unit-after-rounding.js:
```diff
@@ -0,0 +1,17 @@
+TemporalHelpers.assertDuration(almostWeek.round({
+    largestUnit: "weeks",
+    smallestUnit: "days",
+    relativeTo: "2020-01-01"
```
Ditto.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/does-not-balance-up-to-weeks-if-largest-unit-is-larger-than-weeks.js:
```diff
@@ -0,0 +1,16 @@
+
+TemporalHelpers.assertDuration(monthAlmostWeek.round({
+    smallestUnit: "days",
+    relativeTo: "2020-01-01"
```
Ditto.

ptomato commented on test/built-ins/Temporal/Duration/prototype/round/rounding-relative-to-date.js:
```diff
@@ -0,0 +1,40 @@
+const yearAndHalf = new Temporal.Duration(0, 0, 0, 547, 12, 0, 0, 0, 0, 0);
+
+TemporalHelpers.assertDuration(days.round({
+    relativeTo: "2019-01-01",
```
Ditto.

ptomato reviewed (approved):
Looks good modulo requested changes - there are a few more places where we should avoid going through the string parsing path if it's not specifically being tested. Thanks!
